### PR TITLE
Fix auth

### PR DESCRIPTION
**Pull request by RSA-encryption**
No longer dies when using http

### CODE REVIEW

**Comment by CrazyHackGUT** [comment]
`$_SERVER['HTTPS'] ?? false` is much simpler. SB++ already requires PHP with null-coalesce operator support ([added in PHP 7.0](https://www.php.net/manual/en/migration70.new-features.php#migration70.new-features.null-coalesce-op)).

File: web/includes/auth/Auth.php [hunk 1]
```diff
@@ -33,7 +33,7 @@ public static function login(int $aid, int $maxlife)
         self::insertNewToken($jti, $secret);
         self::updateLastVisit($aid);
 
-        self::setCookie($token, time() + $maxlife, Host::domain(), ((bool)$_SERVER['HTTPS']) ? true : false);
+        self::setCookie($token, time() + $maxlife, Host::domain(), array_key_exists('HTTPS', $_SERVER) ? (bool)$_SERVER['HTTPS'] : false);
```
Comment (alexandre433) [inline web/includes/auth/Auth.php:36]: You could also use `(bool) $_SERVER['HTTPS'] ?? false`

File: web/includes/auth/Auth.php [hunk 2]
```diff
@@ -57,7 +57,7 @@ public static function logout()
             self::$dbs->execute();
         }
 
-        self::setCookie('', 1, Host::domain(), ((bool)$_SERVER['HTTPS']) ? true : false);
+        self::setCookie('', 1, Host::domain(), array_key_exists('HTTPS', $_SERVER) ? (bool)$_SERVER['HTTPS'] : false);
```
Comment (alexandre433) [inline web/includes/auth/Auth.php:60]: Same here.

**Comment by Hackmastr** [comment]
Thanks! 

This has been fixed in the php81 branch.

**Pull request closed without merging**